src/bundle/Resources/config/services/storages.yaml

```diff
@@ -0,0 +1,10 @@
+services:
+    _defaults:
+        autowire: true
+        autoconfigure: true
+        public: false
+
+    _instanceof:
+        Ibexa\Contracts\Personalization\Storage\DataSourceInterface:
+            tags:
+                - { name: ibexa.personalization.data_source }
```

Comment:
Note that `_instanceof` only takes effect for services declared in the same file, so it's not "autoconfiguration".

Reply:
I forgot about that. I will use `registerForAutoconfiguration() ` method in bundle Extension class.

Reply:
If you expect all services for this interface to be tagged, then it's only natural to `registerForAutoconfiguration()`. If required autoconfiguration can be disabled by the user, so he can always re-adjust the tags.